spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java

```diff
@@ -122,6 +123,12 @@ public SpannerTemplate spannerTemplate(DatabaseClient databaseClient,
 				spannerMutationFactory);
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	public SpannerSqlTemplate spannerSqlTemplate(SpannerTemplate spannerTemplate) {
+		return spannerTemplate.getSpannerSqlTemplate();
```

Comment:
I think this is backwards. You want to create a `SpannerSqlTemplate` bean and then have `SpannerTemplate` depend on it.

Reply:
I'm sorry I haven't clarified this further - my comment became confusing.

After our discussion today it seems that a SpannerSqlTemplate is something that can live on the lower level and actually can expose Spanner types, similar to CqlTemplate in Cassandra. 

Also, as @meltsufin says SpannerSqlTemplate would be the foundation for SpannerTemplate this way.

Reply:
Let's put this to on-hold then.